prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -341,41 +341,61 @@ class DetailedSubmissionHistory(
                 // Check if destination has already been added
                 // if it is increase item counts
                 // otherwise add it to destinations
-                val index = destinations.indexOf(descendantDest)
-                if (index >= 0) {
-                    destinations[index].itemCount += descendantDest.itemCount
-                    destinations[index].itemCountBeforeQualFilter =
-                        destinations[index].itemCountBeforeQualFilter?.plus(
-                            descendantDest.itemCountBeforeQualFilter ?: 0
-                        )
-                } else {
-                    destinations += descendantDest
+                with(destinations) {
+                    val index = indexOfFirst {
+                        it.organizationId == descendantDest.organizationId && it.service == descendantDest.service
+                    }
+                    if (index >= 0) {
```

Comment:
index is the item index which starts at 1. There was a big effort a while back to standardize the code to use 1 instead of 0, so the database and API responses match.

Reply:
>=0 is right as the index search in line 345 can return a 0.

Reply:
If it returns a 0, wouldn't that mean we found the destination at the first position in the list? I guess my question is more about why we are adding a new destination when we find an existing one at index `0`, instead of only adding one if line 345 returns `-1`

Reply:
This was updated to use firstOrNull, but the if statement  was including 0, so if it returns 0 it would update that destination not add a new one.

Reply:
Oh wow, with fresh eyes I'm seeing that again and it makes perfect sense, I think I'm just used to seeing `index > -1` and got confused, that was my bad 😅